pandas/tests/indexes/datetimes/test_to_period.py

```diff
@@ -75,6 +77,22 @@ def test_to_period_monthish(self):
         with pytest.raises(ValueError, match=INVALID_FREQ_ERR_MSG):
             date_range("01-Jan-2012", periods=8, freq="EOM")
 
+    def test_to_period_infer(self):
+        # https://github.com/pandas-dev/pandas/issues/33358
+        rng = date_range(
+            start="2019-12-22 06:40:00+00:00",
+            end="2019-12-22 08:45:00+00:00",
+            freq="5min",
+        )
+
+        # Using simple filter because we are not checking for the warning here
+        warnings.simplefilter("ignore", UserWarning)
```

Comment:
what is the warning from? this would not be set like this
use assert_produces_warning(None):
    warnings.simplefiler....

if you really need to

Reply:
> what is the warning from? this would not be set like this

The warning raises when a user is attempting to convert a date_range to period, if the date_range have a timezone it will drop it in the conversion, the warning let the user know about that:

https://github.com/pandas-dev/pandas/blob/3cca07c8a5903d3749d478312f4e6d8f12b2ee20/pandas/core/arrays/datetimes.py#L1085-L1090

> use assert_produces_warning(None):
> warnings.simplefiler....
> 
> if you really need to

Sure! will fix.

Reply:
i c, ok then